### PR TITLE
Deprecate versionId

### LINKED ISSUE #812: Deprecate versionId
After #810, none of our code will check `versionId`, as it is an incomplete means of getting the correct versioning behavior. In 2.0, we would then be able to remove references to `versionId` from:

* [x] standard: release-schema.json
* [x] standard: meta-schema.json
* [x] standard: meta-schema-patch.json
* [x] standard: make_versioned_release_schema.py

As part of the deprecation, we'd need to update:

* [x] ocdskit: schema_report.py

### PR DESCRIPTION
**Pull request by yolile**
closes #812

### CODE REVIEW

File: schema/meta-schema.json [hunk 1]
```diff
@@ -245,7 +245,11 @@
     },
     "versionId": {
       "type": "boolean",
-      "default": false
+      "default": false,
+      "deprecated": {
```
Comment (yolile) [inline schema/meta-schema.json]: @jpmckinney I'm not sure if I should remove the field entirely or just adding the deprecated object

Comment (jpmckinney) [reply schema/meta-schema.json]: Remove the field entirely. We only use `deprecated` in the actual schema, not the metaschema.

**Comment by jpmckinney** [comment]
Changelog entry is missing.

**Comment by yolile** [comment]
> Changelog entry is missing.

added

**Comment by jpmckinney** [comment]
I've also removed the note linking to the issue. I'll merge when tests pass.

**Pull request merged**